realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -299,6 +318,12 @@ private boolean categorizeClassElements() {
                         return false;
                     }
 
+                    // A field cannot be nullable and primary key. Currently only String can be nullable
+                    if (Utils.isString(variableElement)
+                            && nullableFieldNames.contains(variableElement)) {
+                        nullableFieldNames.remove(variableElement);
```

Comment:
Is this correct. You are looping the variable elements and are inside a if/else. This particular variable element could not have been added yet?